hazelcast-client/src/main/java/com/hazelcast/client/connection/nio/ClientWriteHandler.java

```diff
@@ -69,28 +66,15 @@ public void handle() throws Exception {
     }
 
     private void writeBuffer() throws IOException {
-        while (buffer.hasRemaining() && lastMessage != null) {
-            boolean complete = lastMessage.writeTo(buffer);
-            if (complete) {
+        while (lastMessage != null) {
+            socketChannel.write(lastMessage);
```

Comment:
@ihsandemir: so it disables "smart-batching", is it what you are saying? 
it makes sense as copying a few bytes and there is cheaper than a system call.